admin/app/components/solidus_admin/ui/table/component.rb

```diff
@@ -136,6 +139,90 @@ def render_data_cell(cell, data)
     content_tag(:td, content_tag(:div, cell, class: "flex items-center gap-1.5"), class: "py-2 px-4 h-10 vertical-align-middle leading-none")
   end
 
+  def search_form_options
+    {
+      url: @search_page_url,
+      method: :get,
+      class: "h-14 p-2 bg-white border-b border-gray-10 relative hidden",
+      data: {
+        "#{stimulus_id}_target": "searchForm"
+      }
+    }
+  end
+
+  def search_field_options
+    {
+      class: "border-2 border-gray-100 bg-white h-10 px-10 rounded-lg text-sm focus:outline-none flex-grow",
+      placeholder: "Search",
+      data: {
+        "#{stimulus_id}_target": "searchField"
+      }
+    }
+  end
+
+  def render_search_form
+    form_with(**search_form_options) do |form|
+      tag.div class: "flex justify-between items-center w-full" do
+        render_search_bar(form) + render_cancel_button
+      end
+    end
+  end
+
+  def render_search_bar(form)
+    tag.div class: "relative flex items-center flex-grow" do
+      render_search_button +
+        form.text_field("q[#{@search_key}]", search_field_options) +
+        render_close_button
+    end
+  end
+
+  def render_search_button
+    @button_component.new(
+      tag: :button,
+      type: :submit,
+      disabled: true,
+      class_name: "left-0 ml-0 border-none bg-transparent absolute hover:bg-transparent hover:text-gray-700",
+      icon_tag: icon_tag("search-line", class: "text-gray-600 h-4 w-4 fill-current"),
+      scheme: :ghost,
+      "aria-label": "Search"
```

Comment:
Same here for i18n.

Reply:
👍 for this specific one, I also included an `"aria-disabled": true` to reflect the search button status.